frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1853,7 +1917,20 @@ function ReaderFooter:genAllFooterText()
             table.insert(info, BD.wrap(text))
         end
     end
-    return table.concat(info, BD.wrap(separator))
+    -- with # as separator, replace "  #" with "  "
+    local retval = table.concat(info, BD.wrap(separator))
+    local pattern = "  " .. BD.wrap(separator) -- two spaces
+    local dummy, remove_pos = retval:find("  " .. BD.wrap(separator)) -- two spaces
+    if separator ~= " " and remove_pos then
+        retval = retval:gsub(pattern, "  ") -- two spaces
+    end
+    -- with # as replace "#  " with "  "
+    pattern = BD.wrap(separator) .. "  "
+    remove_pos = retval:find(BD.wrap(separator) .. "  ") -- two spaces
+    if separator ~= " " and remove_pos then
+        retval = retval:gsub(pattern, "  ") -- two spaces
+    end
+    return retval
```

Comment:
No real comment, except about this, which looks complicated and I don't really understand, but I can't really focus at this time of day :/ But it smells not like roses :)

Reply:
I'm not sure why/when you get doublespaces at start/end of items and I guess you want to remove the separator.
Might be cleaner to loop items in the `info` table, and merge/trim them (still not understand why/when) in there, so you don't have to deal with `BD.wrap(separator) ` at all, and just keep doing the table.concat() at the end.

Reply:
We could get double spaces before or after a separator if the user enters 'custom text ' consisting of spaces, i.e if he wants the custom text as a spacer.

Reply:
You can see the results (without cleaning in the first image ).

Reply:
I have made the other images (after 1 and 2 with not separators  set!)

Reply:
OK. Then I'll leave this to the generator itself, instead of the above big stuff that is of no use for all the other generators (and complicate the otherwise simple previous code).
Something like:
```lua
local text, merge = gen(self) -- only your new one would return merge other than nil
[...]
if merge then
   info[#info] = info[#info] + text
   prev_had_merge = true
elseif prev_had_merge then
   info[#info] = info[#info] + text
   prev_had_merge = false
else
   table.insert(info, BD.wrap(text))
end
```